cognite/neat/rules/exporters/_rules2yaml.py

```diff
@@ -0,0 +1,93 @@
+import json
+import warnings
+from pathlib import Path
+from typing import Literal, get_args
+
+import yaml
+
+from cognite.neat.rules._shared import Rules
+from cognite.neat.rules.models._rules.base import RoleTypes
+
+from ._base import BaseExporter
+
+
+class YAMLExporter(BaseExporter[str]):
+    """Export rules to YAML.
+
+    Args:
+        format: Whether to output the rules as multiple or a single YAML file(s). Defaults to "multiple".
+
+    The following formats are available:
+
+    - "single": A single YAML file will containe the entire rules.
+    - "multiple": Each section in the rules (e.g. same as sheets in the spreadsheet exporter) will be exported to
+                    a separate YAML file.
+    .. note::
+
+        YAML files are typically used for storing rules when checked into version control systems, e.g., git-history.
+        The advantage of using YAML files over Excel is that tools like git can show the differences between different
+        versions of the rules.
+
+    """
+
+    Files = Literal["single", "multiple"]
+    Format = Literal["yaml", "json"]
+
+    file_option = get_args(Files)
+    format_option = get_args(Format)
+
+    def __init__(self, files: Files = "multiple", output: Format = "yaml", output_role: RoleTypes | None = None):
+        if files not in self.file_option:
+            raise ValueError(f"Invalid files: {files}. Valid options are {self.file_option}")
+        if output not in self.format_option:
+            raise ValueError(f"Invalid output: {output}. Valid options are {self.format_option}")
+        self.files = files
+        self.output = output
+        self.output_role = output_role
+
+    def export_to_file(self, filepath: Path, rules: Rules) -> None:
+        """Exports transformation rules to YAML/JSON file(s)."""
+        if self.files == "single":
+            if filepath.suffix != f".{self.output}":
+                warnings.warn(f"File extension is not .{self.output}, adding it to the file name", stacklevel=2)
+                filepath = filepath.with_suffix(f".{self.output}")
+            filepath.write_text(self.export(rules))
+        elif self.files == "multiple":
+            if filepath.is_file():
+                raise FileExistsError(f"{filepath} is a file, cannot export to a directory")
+            filepath.mkdir(parents=True, exist_ok=True)
+            rules = self._convert_to_output_role(rules, self.output_role)
+
```

Comment:
let's make a placeholder for method that will split YAML into multiple one according to the upper comment.